File: src/atcoder/abc310/abc310_b.swift
import Foundation

let line = readLine()!.split(separator: " ").map { Int($0)! }
let n = line[0]

var prices = [Int]()
var functions = [Set<Int>]()

for _ in 0..<n {
  let line = readLine()!.split(separator: " ").map { Int($0)! }
  prices.append(line[0])
  let cnt = line[1]
  var function = Set<Int>()
  for j in 0..<cnt {
    function.insert(line[j+2])
  }
  functions.append(function)
}

for i in 0..<n {
  for j in 0..<n {
    if prices[i] > prices[j] && functions[i].isSubset(of: functions[j]) {
      print("Yes")
      exit(0)
    }

    if prices[i] >= prices[j] && functions[i].isStrictSubset(of: functions[j]) {
      print("Yes")
      exit(0)
    }
  }
}
print("No")

File: src/atcoder/abc310/abc310_b.scala
import java.util.Scanner
import scala.collection.mutable.Seq
import scala.collection.mutable.ArrayBuffer
import scala.collection.mutable.Set

object Main extends App {
  val sc = new Scanner(System.in)
  val n = sc.nextInt
  val m = sc.nextInt

  val prices = ArrayBuffer[Int]()
  val functions = ArrayBuffer[Set[Int]]()

  for (i <- 0 until n) {
    prices += sc.nextInt
    val cnt = sc.nextInt
    val function = Set[Int]()
    for (j <- 0 until cnt) {
      function += sc.nextInt
    }
    functions += function
  }

  for {
    i <- 0 until n
    j <- 0 until n
  } {
    if (
      prices(i) > prices(j) &&
      functions(i).subsetOf(functions(j))
    ) {
      println("Yes")
      sys.exit(0)
    }

    if (
      prices(i) >= prices(j) &&
      functions(i).subsetOf(functions(j)) &&
       (functions(j) -- functions(i)).size > 0
    ) {
      println("Yes")
      sys.exit(0)
    }
  }
  println("No")
}